### PR TITLE
MM 43666 - request bussiness email

### PR DESCRIPTION
**Pull request by pvev**
#### Summary
This is the server-side changes in charge of process the verification of the user and customer emails, making sure that those are valid business emails and if they are not, then request a valid business email.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-43666

#### Related Pull Requests
enterprise: https://github.com/mattermost/enterprise/pull/1215
mattermost-webapp: https://github.com/mattermost/mattermost-webapp/pull/10405
customer-web-server: https://github.com/mattermost/customer-web-server/pull/790

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by pvev** [comment]
@marianunez @neallred the unit tests need to be improved and some checks fixes, but I'd highly appreciate early feedback for this changes. Thanks!

File: api4/cloud.go [hunk 1]
```diff
@@ -161,6 +178,48 @@ func requestCloudTrial(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write(json)
 }
 
+func validateBusinessEmail(c *Context, w http.ResponseWriter, r *http.Request) {
+	if c.App.Channels().License() == nil || !*c.App.Channels().License().Features.Cloud {
+		c.Err = model.NewAppError("Api4.validateBusinessEmail", "api.cloud.license_error", nil, "", http.StatusForbidden)
+		return
+	}
+
+	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleReadBilling) {
```
Comment (marianunez) [inline api4/cloud.go:187]: They should have permissions to make changes to the subscriptions if they are going through the trial process

```suggestion
	if !c.App.SessionHasPermissionTo(*c.AppContext.Session(), model.PermissionSysconsoleWriteBilling) {
```

**Review by neallred — APPROVED** [review]

File: api4/cloud.go [hunk 2]
```diff
@@ -146,7 +149,21 @@ func requestCloudTrial(c *Context, w http.ResponseWriter, r *http.Request) {
...
 
-	changedSub, err := c.App.Cloud().RequestCloudTrial(c.AppContext.Session().UserId, currentSubscription.ID)
+	// check if the email needs to be set
+	bodyBytes, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		c.Err = model.NewAppError("Api4.requestCloudTrial", "api.cloud.app_error", nil, err.Error(), http.StatusInternalServerError)
+		return
+	}
+	// this value will not be empty when both emails (user admin and CWS customer) are not business email and
+	// we need to request a new email from the user via the request business email modal
```
Comment (marianunez) [inline api4/cloud.go:159]:
```suggestion
	// a new business email was provided via the request business email modal 
```

**Review by marianunez — APPROVED** [review]

**Pull request merged**